### PR TITLE
fix(demo): line days chart labels visibility

### LINKED ISSUE #473: [BUG] [LineDaysChart] X axis labels are not visible
# 🐞 Bug report

### Description

`LineDaysChart` component demo has broken X axis. X axis labels are not visible.

### Reproduction

1. Open `LineDaysChart` demo
2. Look at the X Axis.

### Expected behavior

There should be X axis labels visible.

### Additional context

taiga-ui dev_components_line-days-chart

### PR DESCRIPTION
**Pull request by bioN1ck**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Refactoring
- [ ] Code style update
- [ ] Build or CI related changes
- [ ] Documentation content changes

Closes #473 

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: projects/demo/src/modules/charts/line-days-chart/examples/1/index.ts [hunk 1]
```diff
@@ -27,19 +29,22 @@ export class TuiLineDaysChartExample1 {
 
     readonly maxLength: TuiDayLike = {month: 12};
 
-    constructor(@Inject(TUI_MONTHS) private readonly months: readonly string[]) {}
+    readonly xStringify$: Observable<TuiStringHandler<TuiDay>> = this.months$.pipe(
+        map(months => {
```
Comment (waterplea) [inline projects/demo/src/modules/charts/line-days-chart/examples/1/index.ts:33]: nit: Unnecessary `{}` and `return`

File: projects/demo/src/modules/charts/line-days-chart/examples/1/index.ts [hunk 2]
```diff
@@ -27,19 +29,22 @@ export class TuiLineDaysChartExample1 {
...
+            return ({month, day}) => `${months[month]}, ${day}`;
+        }),
+    );
+
+    constructor(@Inject(TUI_MONTHS) readonly months$: Observable<readonly string[]>) {}
 
-    get labels(): readonly string[] {
-        return this.computeLabels(this.range, this.months);
+    get labels$(): Observable<readonly string[]> {
```
Comment (waterplea) [inline projects/demo/src/modules/charts/line-days-chart/examples/1/index.ts:40]: This creates a new `Observable` on every change detection cycle. `computeLabels` is a function with memoization (it uses `@tuiPure` decorator) so you need to move `Observable` creation there to get rid of unnecessary recalculations and improve performance.

File: projects/demo/src/modules/charts/line-days-chart/examples/2/pipe.ts [hunk 3]
```diff
@@ -1,21 +1,29 @@
 import {Inject, Pipe, PipeTransform} from '@angular/core';
 import {TuiDay, TuiDayRange, TuiMonth} from '@taiga-ui/cdk';
 import {TUI_MONTHS} from '@taiga-ui/core';
+import {Observable, of} from 'rxjs';
+import {map} from 'rxjs/operators';
 
 // @dynamic
 @Pipe({
-    name: 'labels',
+    name: 'labels$',
```
Comment (waterplea) [inline projects/demo/src/modules/charts/line-days-chart/examples/2/pipe.ts:9]: I don't think we should add `$` to pipe's name, it looks confusing as if it's not a pipe but a variable when you see it in template.

File: projects/demo/src/modules/charts/line-days-chart/examples/2/pipe.ts [hunk 4]
```diff
@@ -1,21 +1,29 @@
...
-    transform({from, to}: TuiDayRange): readonly string[] {
+    transform({from, to}: TuiDayRange): Observable<readonly string[]> {
         const length = TuiDay.lengthBetween(from, to);
 
         if (length > 90) {
-            return Array.from(
-                {length: TuiMonth.lengthBetween(from, to) + 1},
-                (_, i) => this.months[from.append({month: i}).month],
+            return this.months$.pipe(
+                map(months => {
```
Comment (waterplea) [inline projects/demo/src/modules/charts/line-days-chart/examples/2/pipe.ts:21]: nit: Unnecessary `{}` and `return`

File: projects/demo/src/modules/charts/line-days-chart/line-days-chart.component.ts [hunk 5]
```diff
@@ -65,33 +67,38 @@ export class ExampleTuiLineDaysChartComponent {
 
     readonly value = this.valueVariants[0];
 
-    readonly labels = Array.from({length: 3}, (_, i) => this.months[i]);
+    readonly labels$: Observable<readonly string[]> = this.months$.pipe(
```
Comment (waterplea) [inline projects/demo/src/modules/charts/line-days-chart/line-days-chart.component.ts:70]: nit: unnecessary type and `{}` + `return`

**Review by waterplea — COMMENTED** [review]
Thank you for this PR, left a few small comments, otherwise LGTM :)

File: projects/demo/src/modules/charts/line-days-chart/examples/1/index.ts (hunk 2 above)
Comment (bioN1ck) [reply projects/demo/src/modules/charts/line-days-chart/examples/1/index.ts]: Check it out again, please

**Review by waterplea — APPROVED** [review]

**Review by vladimirpotekhin — APPROVED** [review]

**Pull request merged**